rfcs/0064-documentation-format.md

```diff
@@ -0,0 +1,142 @@
+---
+feature: documentation-format
+start-date: 2019-12-31
+author: Silvan Mosberger (infinisil)
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+The Nix community wants to move away from using Docbook as our documentation format and replace it with something else. However it is unclear what it should be replaced with. This RFC gives a concrete process for determining the new documentation format. It does NOT say what format should be used.
+
+# Motivation
+[motivation]: #motivation
+
+There's been enough bike-shedding over the documentation format to use. We should finally settle this debate by deciding and committing on a single format such that we can move forward and improve our documentation situation.
+
+# Detailed design
+[design]: #detailed-design
+
+The process for determining the doc format is as follows:
+- A set of requirements for the doc format is decided through the RFC discussion
+- Doc format candidates are collected and evaluated to see if they fulfil the requirements.
+- A short objective overview of each valid candidate format is written, along with their advantages/disadvantages
+- The RFC is accepted
+- A [Discourse](https://discourse.nixos.org/) post is created with these overviews, along with a poll such that people can vote on the formats they prefer. This poll will be open to the whole community and should be advertised as such
+- Whatever format wins in the poll is chosen as the new default documentation format. If later it is discovered that the winner is infeasible for any reason, e.g. if it doesn't meet the requirements after all, the format on second place is chosen instead, and so on.
+
+## Poll
+
+The poll is of the following form:
+- Multiple-choice, allowing people to select all formats they agree with
+- Results are only shown when the poll is closed for it to not be influenced by non-final tallies
+- Answers can be changed while the poll is still active, allowing people to discuss about formats and change their opinion (this is not optional in Discourse)
+- It runs for 1 month to give enough time for less-active people to see it
+- Who voted for which options is made public (Only possible with bar chart in Discourse) TODO: Do we want this or not? Why would we?
+
+## Requirements
+
+- Can be converted to HTML and man pages
+- Inter-file references for being able to link to options from anywhere
+- Ability to create link anchors to most places such that we can link to e.g. paragraphs
+- Errors are easily and quickly detectable, e.g. with a fast and good processor, a live-view, or highlighting editor plugins
+- Is decently fast to fully generate, in the range of 10 seconds for the full documentation on an average machine
```

Comment:
- supports syntax highlighting (of Nix)
- wide editor integration (this could be added to the errors bullet point)
- active community supporting the tooling infrastructure
- good conversion story from docbook
- ideally a good search integration
- low work cost for integration (this one is really important as docbook fails hard)

Reply:
Nice, what do you mean by the last point though? That we don't need to write too many customizations and supporting code to make it work?